packages/client/lib/service/txpool.ts

```diff
@@ -282,6 +282,9 @@ export class TxPool {
         `Tx gaslimit of ${tx.gasLimit} exceeds block gas limit of ${block.gasLimit} (exceeds last block gas limit)`
       )
     }
+
+    // Set state root to latest block so that account balance is correct when doing balance check
+    await this.vm.stateManager.setStateRoot(block.stateRoot)
```

Comment:
Doesn't this add a race condition if we are running blocks at the same time? (Or is this a different VM than we use for running the blocks?)

Reply:
Looks like it's the same VM from looking at the txpool constructor.  I can copy it here and pull state that way since the underlying db should be the same.